build-tools/webpack/generate-chunks-map-plugin.js

```diff
@@ -18,18 +18,20 @@ class GenerateChunksMapPlugin {
 			// Generate chunks map
 			const { chunks } = compilation;
 
-			const chunksMap = chunks.reduce( ( map, chunk ) => {
-				const files = chunk.files;
-				const name = files.find( ( file ) => /\.js$/.test( file ) ) || files[ 0 ];
-				const modules = [ ...chunk.modulesIterable ]
+			const chunksMap = {};
+			for ( const chunk of chunks ) {
+				// This logic assumes there is only one `.js`. If there are more than one `.js` file linked to a chunk,
+				// this will be non deterministic as `chunk.files` iteration order is not guaranteed.
+				const name = Array.from( chunk.files ).find( ( file ) => /\.js$/.test( file ) );
+				if ( ! name ) continue;
```

Comment:
In Webpack 4, if a `chunk` doesn't contain a `.js` file we just pick the first file in the `chunk`. This caused some entries like `chunksMap["file.<hash>.css"]=[]`.

In Webpack 5, `chunk.files` is a Set, so there is no "first" file (i.e. the result will be non deterministic). I've changed the logic to skip those `chunks` instead.

It shouldn't affect the end result, as the entries without `.js` files were empty anyway (cc @yuliyan to confirm)

Reply:
> It shouldn't affect the end result, as the entries without .js files were empty anyway.
That's correct. Such entries have no use.